src/utils/deleteEmptyValuesFromObject.ts

```diff
@@ -0,0 +1,16 @@
+import isEmpty from 'lodash/isEmpty';
+import isNil from 'lodash/isNil';
+
+type QueryObject = Record<string, string | string[] | null | undefined>;
+
+const deleteEmptyValuesFromObject = (
+  obj: QueryObject
+): Record<string, string | string[]> => {
+  return Object.entries(obj).reduce((acc, [key, value]) => {
+    // If value is null, undefined or empty object/array, don't append
+    if (isNil(value) || isEmpty(value)) return acc;
+    return { ...acc, [key]: value };
+  }, {});
+};
```

Comment:
Isn't this already in lodash: `_.pickBy(obj);`? https://stackoverflow.com/a/35871405/784642

Reply:
Without lodash that reducer for object entries that you have written would be hot! 👍 ..But then you wouldn't have isNil either :P There are several alternatives introduced here: https://stackoverflow.com/questions/286141/remove-blank-attributes-from-an-object-in-javascript

Reply:
and note that there are those `_.omit(obj, _.isUndefined)` or `_.omitBy(obj, _.isNil)` alternatives in lodash also! Since you have used isNil, maybe the latter is better?

Reply:
Nice. I think `_.omitBy(object, _.isNil)` is the way to go as we want to get rid of empty arrays too.

Reply:
Sorry, we need both _.isNil and isEmpty

Reply:
`_.omitBy(object, (v) => _.isNil(v) || _.isEmpty(v))`?